### ISSUE TITLE
RuntimeError: memory format option is only supported by strided tensors

### ISSUE THREAD

**Issue by JiaerXia** [Dao-AILab/flash-attention#782]
Following the usage, I got the following error when calling **flash_attn_func function**.

**RuntimeError: memory format option is only supported by strided tensors**

However, I have checked all the tensor are strided.
![1706008766239](https://github.com/Dao-AILab/flash-attention/assets/113883823/2d6965bd-1762-4fe6-a277-d90bf53551e5)
I hope you can get help to solve this problem.

**Comment by tridao** [comment]
Can you add a short script to reproduce the error?

**Comment by JiaerXia** [comment]
I'm trying to rewrite the multiheadattention to speed up the clip model, like the code in this [repo](https://github.com/OrangeSodahub/Openclip-flashattn/blob/4731c98253fd37e77fdb2bf762d4acb863050129/modeling/flash_attn/flash_attention.py#L13).
I also tried to use the MHA module of flash_attn, but the error remained.

**Comment by JonasGeiping** [comment]
I have also some trouble with this error in recent days. It appears when switching pytorch nightlies from `2.3.0.dev20240122`  to `2.3.0.dev20240123` (or newer), so likely related to updates in PyTorch - but it is hard for me to exactly figure out which PyTorch commit is responsible.  

No easy repro from me either, sorry, but the error occurs on every invocation of flash attention when doing forward passes with the huggingface implementation of Llama-2.

**Comment by JiaerXia** [comment]
Thank you for sharing this information, I am indeed using the nightliest pytorch and believe this is the source of the problem.

**Comment by JiaerXia** [comment]
> I have also had some trouble with this error in recent days. It appears when switching pytorch nightlies from `2.3.0.dev20240122` to `2.3.0.dev20240123` (or newer), so likely related to updates in PyTorch - but it is hard for me to exactly figure out which PyTorch commit is responsible.
> 
> No easy repro from me either, sorry, but the error occurs on every invocation of flash attention when doing forward passes with the huggingface implementation of Llama-2.

Thank you for sharing this information, I am indeed using the nightliest pytorch and believe this is the source of the problem.

**Comment by michael-sandoval** [comment]
Seeing similar behavior with 2.1.2, 2.2.0 when trying to run the benchmark script (albeit with ROCm installations).

- Stable release of 2.2.0 w/ ROCm 5.7 sees this error
- Stable release of 2.1.2 w/ ROCm 5.6 works fine
- Nightly wheel `2.2.0.dev20231010+rocm5.7` works fine
- Nightly wheel `2.2.0.dev20231106+rocm5.7` also works fine

Unsure which PyTorch commit between November 6th 2023 and the stable release of >=2.2.0 broke things.

**Comment by tridao** [comment]
Idk much about the ROCm version, you might have better luck asking on their repo.

**Comment by michael-sandoval** [comment]
Right, understood -- just thought I'd add that context here since the problem doesn't seem to be isolated to ROCm installations

**Comment by yiyi20209** [comment]
I'm facing same issues. Any update on this issue would be really appreciated :)

**Comment by npz7yyk** [comment]
Met the same problem with torch2.2.1+cu121. Switched to torch.2.3.0 and it was fixed. Likely related to pytorch.
